File: src/main/java/com/mhl/mycompanybackend/model/UserData.java
package com.mhl.mycompanybackend.model;


import javax.persistence.*;

@Entity
@Table(name = "user_data")
public class UserData {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    private String firstname;
    private String lastname;
    private String middlename;
    private String phone_number;

    @OneToOne(mappedBy = "userData")
    private Users user;

    public UserData(String firstname, String lastname, String middlename, String phoneNumber) {
        this.firstname = firstname;
        this.lastname = lastname;
        this.middlename = middlename;
        phone_number = phoneNumber;
    }

    public UserData() {
    }

    public String getFirstname() {
        return firstname;
    }

    public void setFirstname(String firstname) {
        this.firstname = firstname;
    }

    public String getLastname() {
        return lastname;
    }

    public void setLastname(String lastname) {
        this.lastname = lastname;
    }

    public String getMiddlename() {
        return middlename;
    }

    public void setMiddlename(String middlename) {
        this.middlename = middlename;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public Long getId() {
        return id;
    }

    public String getPhone_number() {
        return phone_number;
    }

    public void setPhone_number(String phone_number) {
        this.phone_number = phone_number;
    }
}

File: src/main/kotlin/com/mhl/mycompanybackend/model/UserData.kt
package com.mhl.mycompanybackend.model

import javax.persistence.*

@Entity
@Table(name = "user_data")
open class UserData {
    @JvmField
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    var id: Long? = null
    @JvmField
    var firstname: String? = null
    var lastname: String? = null
    var middlename: String? = null
    var phone_number: String? = null

    @OneToOne(mappedBy = "userData")
    private val user: Users? = null

    constructor(firstname: String?, lastname: String?, middlename: String?, phoneNumber: String?) {
        this.firstname = firstname
        this.lastname = lastname
        this.middlename = middlename
        phone_number = phoneNumber
    }

    constructor()
}
